include/crypto/sha.h

```diff
@@ -12,10 +12,13 @@
 # define OSSL_CRYPTO_SHA_H
 # pragma once
 
-# include <openssl/opensslconf.h>
+# include <openssl/sha.h>
 
+# ifndef OPENSSL_NO_DEPRECATED_3_0
 int sha512_224_init(SHA512_CTX *);
 int sha512_256_init(SHA512_CTX *);
 int ossl_sha1_ctrl(SHA_CTX *ctx, int cmd, int mslen, void *ms);
+# endif
```

Comment:
Not sure why you find it necessary to deprecate internal functions...

Reply:
Me neither. The CI run `enable_non-default_options` was failing like this:
```
 In file included from crypto/evp/e_des3.c:22:
include/crypto/sha.h:18:21: error: unknown type name 'SHA512_CTX'; did you mean 'SHA512_ASM'?
   18 | int sha512_224_init(SHA512_CTX *);
      |                     ^~~~~~~~~~
      |                     SHA512_ASM
include/crypto/sha.h:19:21: error: unknown type name 'SHA512_CTX'; did you mean 'SHA512_ASM'?
   19 | int sha512_256_init(SHA512_CTX *);
      |                     ^~~~~~~~~~
      |                     SHA512_ASM
include/crypto/sha.h:20:20: error: unknown type name 'SHA_CTX'; did you mean 'SCT_CTX'?
   20 | int ossl_sha1_ctrl(SHA_CTX *ctx, int cmd, int mslen, void *ms);
      |                    ^~~~~~~
      |                    SCT_CTX
```
so I thought this was due to deprecation, but it could also have been due to a missing `#include` which I added meanwhile, so let's see how it works without that guard...

Reply:
Right, the `# ifndef OPENSSL_NO_DEPRECATED_3_0` was not really needed, so removed it.